dashboard/app/models/user.rb

```diff
@@ -1610,26 +1613,41 @@ def can_access_most_recently_assigned_script?
     !script.pilot? || script.has_pilot_access?(self)
   end
 
+  # Query to get the user_script the user made the most recent progress
+  # in.
   def user_script_with_most_recent_progress
     user_scripts.
     where("last_progress_at").
     order(last_progress_at: :desc).
     first
   end
 
+  # Get script object of the user_script the user made the most recent
+  # progress in.
   def script_with_most_recent_progress
     user_script_with_most_recent_progress.script
   end
 
+  # Check if the user's most recently-assigned script is the same one
+  # that they've most recently made progress in.
   def most_recent_progress_in_recently_assigned_script?
     script_with_most_recent_progress == most_recently_assigned_script
   end
 
+  # Check if the user has been assigned a new script since their most
+  # recent progress in a script.
   def last_assignment_after_most_recent_progress?
     most_recently_assigned_user_script[:assigned_at] >=
     user_script_with_most_recent_progress[:last_progress_at]
   end
 
+  # Check if the user's most recently assigned script is associated with at least
+  # 1 live section they are enrolled in.
+  def most_recent_assigned_script_in_live_section?
+    recent_assigned_script_id = most_recently_assigned_script.id
```

Comment:
Can `most_recently_assigned_script` be nil?

Reply:
Yes but it would be caught first by user.can_access_most_recently_assigned_script? (covered by Scenario 1 where the student is in no sections).